corehq/apps/domain/models.py

```diff
@@ -822,6 +823,11 @@ def save_snapshot(self, share_reminders, copy_by_id=None):
             except NameUnavailableException:
                 return None
             copy.is_snapshot = True
+            for snapshot in self.snapshots():
```

Comment:
Rather than iterating through all snapshots could you not just get any with `snapshot_head = True`. Should be easy to get from ES.